tools/unit/src/main/kotlin/com/datadog/tools/unit/assertj/JsonObjectAssert.kt

```diff
@@ -433,20 +433,49 @@ class JsonObjectAssert(actual: JsonObject) :
         expectedValue.forEach {
             val value = it.value
             val key = it.key
-            when (value) {
-                is Boolean -> hasField(key, value)
-                is Int -> hasField(key, value)
-                is Long -> hasField(key, value)
-                is Float -> hasField(key, value)
-                is Double -> hasField(key, value)
-                is String -> hasField(key, value)
-                is Date -> hasField(key, value.time)
-                is BigInteger -> hasField(key, value)
-            }
+            checkHasField(value, key)
         }
         return this
     }
 
+    private fun JsonObjectAssert.checkHasField(
+        value: Any?,
+        key: String
```

Comment:
It's ok 👌 , but a bit weird as: `key`, `value` feels more natural than reversed: `value`, `key`.

Reply:
Oh...that was not needed anymore...I forgot to remove and move everything in the main function. Nice spotting ;)